cpp/src/arrow/util/decimal_test.cc

```diff
@@ -985,6 +986,45 @@ TEST(Decimal128Test, Divide) {
   }
 }
 
+TEST(Decimal128Test, Rescale) {
+  ASSERT_EQ(Decimal128(11100), Decimal128(111).Rescale(0, 2).ValueOrDie());
```

Comment:
Nit: we've got `ASSERT_OK_AND_EQ` for this. For example:
```c++
ASSERT_OK_AND_EQ(Decimal128(11100), Decimal128(111).Rescale(0, 2));
```

Reply:
Changed accordingly. Thank you!